### PR TITLE
UI & CORE: Add banners to Quicksearch, Search landing page and merchandising

### PR DESCRIPTION
**Pull request by rallu**
* Refactor type locations in core
* Add banners to
  * Quicksearch
  * Search landing page
  * Merchandising

### CODE REVIEW

File: packages/klevu-ui/package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
   },
   "dependencies": {
     "@floating-ui/dom": "^1.4.5",
-    "@klevu/core": "3.4.0",
+    "@klevu/core": "file:../klevu-core",
```
Comment (nikhil-narayana-klevu) [inline packages/klevu-ui/package.json:47]: is this ok to have here?

File: packages/klevu-ui/src/components/klevu-banner/klevu-banner.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@Component({
+  tag: "klevu-banner",
+  styleUrl: "klevu-banner.css",
+  shadow: true,
+})
+export class KlevuBanner {
+  @Prop() imageUrl?: string
+  @Prop() linkUrl?: string
+  @Prop() altText?: string
+  @Prop() target: "_blank" | "_self" = "_blank"
```
Comment (nikhil-narayana-klevu) [inline packages/klevu-ui/src/components/klevu-banner/klevu-banner.tsx:15]: need to add documentation

File: packages/klevu-ui/storybookAddons/klevuSettings.tsx [hunk 3]
```diff
@@ -39,18 +39,21 @@ export const Tool = memo(function KlevuSettings() {
...
     localStorage.setItem("klevu-language", language ?? "")
+    localStorage.setItem("klevu-use-kmc", useKMC.toString() ?? "true")
     window.location.reload()
   }
 
   const onClose = () => {
     setApiKey(localStorage.getItem("klevu-api-key") || "")
     setUrl(localStorage.getItem("klevu-url") || "")
     setLanguage(localStorage.getItem("klevu-language") || undefined)
+    setUseKMC(localStorage.getItem("klevu-use-kmc") === "false" ? false : true)
```
Comment (nikhil-narayana-klevu) [inline packages/klevu-ui/storybookAddons/klevuSettings.tsx:56]: is this value being used when we do the init? I dont see references to this key in this PR. think you have used the wrong key in `storybook-utils.ts`

**Review by nikhil-narayana-klevu — COMMENTED** [review]
added comments

File: packages/klevu-ui/package.json (hunk 1 above)
Comment (rallu) [reply packages/klevu-ui/package.json:47]: Will change when I know what version we will have.

File: packages/klevu-ui/storybookAddons/klevuSettings.tsx (hunk 3 above)
Comment (rallu) [reply packages/klevu-ui/storybookAddons/klevuSettings.tsx:56]: Good find!

**Review by nikhil-narayana-klevu — APPROVED** [review]

**Pull request merged**